zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -690,10 +691,30 @@ public long getStorageSize() {
          * @throws IOException
          */
         private boolean goToNextLog() throws IOException {
-            if (storedFiles.size() > 0) {
+            while (storedFiles.size() > 0) {
                 this.logFile = storedFiles.remove(storedFiles.size()-1);
-                ia = createInputArchive(this.logFile);
-                return true;
+                try {
+                    ia = createInputArchive(this.logFile);
+                    return true;
+                } catch (StreamCorruptedException ex) {
+                    // The magic number is corrupted. We shouldn't simply skip
+                    // this log file since it might contain transactions that
+                    // have already been acknowledged.
+                    throw ex;
+                } catch (EOFException ex) {
```

Comment:
EOFException could throw if the txn file is corrupted as well, right? For example, the txn file is truncated to only including the magic and version, but not dbid field.

It may not safe to just delete the txn file and move on.

Reply:
txnlog file may not just lose dbid field, but also serval txn bytes blocks. The rest of txn byte blocks are still complete and acknowledged to clients